packages/plugin-commands-outdated/src/outdated.ts

```diff
@@ -289,26 +326,32 @@ export function renderCurrent ({ current, wanted }: OutdatedPackage) {
   return `${output} (wanted ${wanted})`
 }
 
-export function renderLatest (outdatedPkg: OutdatedWithVersionDiff): string {
+export function renderLatest (outdatedPkg: OutdatedWithVersionDiff, withoutChalk = false): string {
   const { latestManifest, change, diff } = outdatedPkg
   if (latestManifest == null) return ''
   if (change === null || (diff == null)) {
     return latestManifest.deprecated
-      ? chalk.redBright.bold('Deprecated')
+      ? withoutChalk ? 'Deprecated' : chalk.redBright.bold('Deprecated')
       : latestManifest.version
   }
 
+  if (withoutChalk) {
+    return diff.flat().join('.')
+  }
+
   return colorizeSemverDiff({ change, diff })
 }
 
-export function renderDetails ({ latestManifest }: OutdatedPackage) {
+export function renderDetails ({ latestManifest }: OutdatedPackage, withoutChalk = false) {
   if (latestManifest == null) return ''
   const outputs = []
   if (latestManifest.deprecated) {
-    outputs.push(wrapAnsi(chalk.redBright(latestManifest.deprecated), 40))
+    const detail = latestManifest.deprecated
+    outputs.push(wrapAnsi(withoutChalk ? detail : chalk.redBright(detail), 40))
```

Comment:
why do you need to wrap for the json output?

Reply:
Sorry, just mistake. [90069a4](https://github.com/pnpm/pnpm/pull/5582/commits/90069a4fa1545f84df758f765869f02c31eed3e2) :+1: